core/src/main/java/org/apache/iceberg/CachingCatalog.java

```diff
@@ -144,14 +144,16 @@ public Table loadTable(TableIdentifier ident) {
       return cached;
     }
 
-    if (MetadataTableUtils.hasMetadataTableName(canonicalized)) {
+    Table table = tableCache.get(canonicalized, catalog::loadTable);
+
+    if (table instanceof BaseMetadataTable) {
+      // Cache underlying table
       TableIdentifier originTableIdentifier =
           TableIdentifier.of(canonicalized.namespace().levels());
       Table originTable = tableCache.get(originTableIdentifier, catalog::loadTable);
 
-      // share TableOperations instance of origin table for all metadata tables, so that metadata
-      // table instances are
-      // also refreshed as well when origin table instance is refreshed.
+      // Share TableOperations instance of origin table for all metadata tables, so that metadata
+      // table instances are also refreshed as well when origin table instance is refreshed.
```

Comment:
I know you're simply reformatting an existing comment, but "also" is redundant here:
"... are also refreshed as well..." -> "... are refreshed as well ...".